### PR TITLE
Update to Commonmark v2

### PR DESCRIPTION
**Pull request by alexanderglueck**
# Description
This pull request makes commonmark-highlighter compatible with league/commonmark v2. By upgrading to v2, support for v1 is dropped. **This is a breaking change**, and therefor I marked the release in the changelog as 3.0.

## Changes: 
- Removed php 7.2 and php 7.3 from the Github Actions tests because commonmark now requires at least php 7.4 (see: https://commonmark.thephpleague.com/2.0/upgrading/integrators/#minimum-php-version)
- Added a changelog entry in the style of the previous commonmark upgrade. This is now a new major version.
- Updated composer.json to be in line with the new commonmark requirements
- Updated the tests

## Changes to tests:
highlight.php produces different output depending on the installed version: 
- Without my changes to the fixtures of the tests: If you run the tests with highlight.php locked at exactly 9.18.1.4 all the tests pass.
- Using the latest version (9.18.1.7) the generated output differs which causes the tests to fail.
**I updated the tests to use the newly generated output.** However, now testing for "prefer-lowest" fails because .4 is installed instead of .7. 

I am looking forward to suggestions on how I can fix this tricky situation (fixtures). 

TLDR on tests: highlight.php 9.18.1.4 generates different output than 9.18.1.7. I updated the fixtures to work with the latest version. Tests fail because of prefer-lowest.

## Closing

Let me know if there is anything else you need.

### CODE REVIEW

**Comment by freekmurze** [comment]
This is awesome, I'll review it soon.

I you feel up to it, you could send a similar PR to [our commonmark-shiki-highlighter package](https://github.com/spatie/commonmark-shiki-highlighter), which provides superior highlighting.

**Comment by freekmurze** [comment]
@alexanderglueck could you make those failing tests pass by updating the versions of requirements of the package?

**Comment by alexanderglueck** [comment]
I increased the minimum required version of highlight.php to 9.18.1.7 and now the tests are only failing (on my branch) due to not being able to push to Scrutinizer. 
So it should probably work on your account.

**Comment by freekmurze** [comment]
@alexanderglueck running the tests... 🤞

**Comment by alexanderglueck** [comment]
You're welcome. Just one more quick question, when can I expect a release so I can change my project to pull in the new version? 😄

**Comment by freekmurze** [comment]
You can expect it in... one minute.

**Comment by alexanderglueck** [comment]
Thank you!
